pos-print/src/main/java/com/clouway/pos/print/adapter/http/ReportService.kt

```diff
@@ -58,17 +57,24 @@ class ReportService(private var factory: PrinterFactory) {
     return Reply.with("Report completed").`as`(Json::class.java).ok()
   }
 
-  private fun adapt(state: String): RegisterState {
+  private fun adaptToState(state: String): RegisterState {
     if (state.equals("CLEAR")) {
       return RegisterState.CLEAR
     }
     return RegisterState.KEEP
   }
 
-  private fun adaptToDate(date: String): LocalDate {
-    return LocalDate.parse(date, DateTimeFormatter.ISO_DATE_TIME)
+  private fun adaptToType(type: String): PeriodType {
+    if (type.equals("SHORT")) {
+      return PeriodType.SHORT
+    }
+    return PeriodType.EXTENDED
+  }
+
+  private fun adaptToDate(date: String): LocalDateTime {
+    return LocalDateTime.parse(date, DateTimeFormatter.ISO_DATE_TIME)
```

Comment:
private val gson = GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create()
did you try it like that ?